fiftyone/zoo/datasets/__init__.py

```diff
@@ -1252,6 +1453,290 @@ def _is_dataset_ready(self, dataset_dir, info, overwrite=False):
+        return tuple(value)
+
+    @property
+    def metadata(self):
+        return self._metadata.copy()
+
+    @property
+    def name(self):
+        return self._name
+
+    @property
+    def url(self):
+        return self._url
+
+    @property
+    def is_remote(self):
+        return True
+
+    @property
+    def author(self):
+        return self._author
+
+    @property
+    def version(self):
+        return self._version
+
+    @property
+    def source(self):
+        return self._source
+
+    @property
+    def license(self):
+        return self._license
+
+    @property
+    def description(self):
+        return self._description
+
+    @property
+    def fiftyone_version(self):
+        return self._fiftyone_version
+
+    @property
+    def tags(self):
+        return self._tags
+
+    @property
+    def supported_splits(self):
+        return self._splits
+
+    @property
+    def supports_partial_downloads(self):
+        return self._supports_partial_downloads
+
+    @property
+    def size_samples(self):
+        return self._size_samples
+
+    def _download_and_prepare(self, dataset_dir, _, split=None):
+        if split is not None:
+            dataset_dir = os.path.dirname(dataset_dir)
+
+        module = self._import_module(dataset_dir)
+        if not hasattr(module, "download_and_prepare"):
+            raise ValueError(
+                f"Module {dataset_dir} has no 'download_and_prepare()' method"
+            )
+
+        kwargs, _ = fou.extract_kwargs_for_function(
+            module.download_and_prepare, self._kwargs
+        )
+        if split is not None:
+            kwargs["split"] = split
+
+        return module.download_and_prepare(dataset_dir, **kwargs)
+
+    def _load_dataset(self, dataset, dataset_dir, split=None):
+        if split is not None:
+            dataset_dir = os.path.dirname(dataset_dir)
+
+        module = self._import_module(dataset_dir)
+        if not hasattr(module, "load_dataset"):
+            raise ValueError(
+                f"Module {dataset_dir} has no 'load_dataset()' method"
+            )
+
+        kwargs, _ = fou.extract_kwargs_for_function(
+            module.load_dataset, self._kwargs
+        )
+        if split is not None:
+            kwargs["split"] = split
+
+        return module.load_dataset(dataset, dataset_dir, **kwargs)
+
+    def _import_module(self, dataset_dir):
+        module_path = os.path.join(dataset_dir, "__init__.py")
+        module_name = os.path.relpath(
+            dataset_dir, fo.config.dataset_zoo_dir
+        ).replace("/", ".")
+        spec = importlib.util.spec_from_file_location(module_name, module_path)
+        module = importlib.util.module_from_spec(spec)
+        # sys.modules[module.__name__] = module
+        spec.loader.exec_module(module)
+        return module
+
+
+def _normalize_ref(url_or_gh_repo):
+    if etaw.is_url(url_or_gh_repo):
+        return url_or_gh_repo
+
+    return "https://github.com/" + url_or_gh_repo
```

Comment:
not sure if I follow the logic here, do we assume that the default host for remote datasets is github?

Reply:
ah got it now!